packages/devtools-move/sdk/baseTaskHelper.ts

```
@@ -0,0 +1,96 @@
+import { getConnection } from './moveVMConnectionBuilder'
+import { RESTClient } from '@initia/initia.js'
+import { Aptos } from '@aptos-labs/ts-sdk'
+import { IOFT } from './IOFT'
+import { OAppOmniGraphHardhat } from '@layerzerolabs/toolbox-hardhat'
+import * as path from 'path'
+
+export interface TaskContext {
+    accountAddress: string
+    privateKey: string
+    chain: string
+    stage: string
+    oAppAddress: string
+    oft: IOFT
+    moveVMConnection: Aptos | RESTClient
+    srcEid: EndpointId
+    lzConfig: OAppOmniGraphHardhat
+    selectedContract: OAppOmniGraphHardhat['contracts'][number]
+    fullConfigPath: string
+}
+
+export interface DeployTaskContext {
+    accountAddress: string
+    privateKey: string
+    chain: string
+    stage: string
+    moveVMConnection: Aptos | RESTClient
+    srcEid: EndpointId
+    lzConfig: OAppOmniGraphHardhat
+    selectedContract: OAppOmniGraphHardhat['contracts'][number]
+    fullConfigPath: string
+}
+
+export async function initializeTaskContext(configPath: string): Promise<TaskContext> {
+    const lzConfig = await getLzConfig(configPath)
+    const moveVMContracts = getMoveVMContracts(lzConfig)
+    const selectedContract = await promptUserContractSelection(moveVMContracts)
+    const network = getNetworkForChainId(selectedContract.contract.eid)
+    const chainName = network.chainName
+    const stage = network.env
+    const accountAddress = getMoveVMAccountAddress(chainName)
+    const moveVMConnection = getConnection(chainName, stage)
+    const moveVMPrivateKey = getMoveVMPrivateKey(chainName)
+    const srcEid = selectedContract.contract.eid
+    const contractName = getContractNameFromLzConfig(srcEid, lzConfig)
+    const oAppAddress = getMoveVMOAppAddress(contractName, chainName, stage)
+    const oft = OFTFactory.create(moveVMConnection, oAppAddress, accountAddress, moveVMPrivateKey, srcEid)
+    const fullConfigPath = path.join(process.cwd(), configPath)
```

Lots of function calls here -- wonder if it is worth putting this into a try catch

---

if any of these fail it should throw an error and stop the program so try catch would not really help